docs/proposals/13-automated-seed-management.md

```diff
@@ -0,0 +1,289 @@
+# Automated Seed Management
+
+Automated seed management involves automating certain aspects of managing seeds in Garden clusters, such as:
+
+* Ensuring that the seeds capacity for shoots is not exceeded.
+* Creating, deleting, and updating seeds declaratively (as "shooted seeds").
+* Auto-scaling seeds upon reaching capacity thresholds.
+
+Implementing the above features would involve changes to various existing Gardener components, as well as perhaps introducing new ones. This document describes these features in more detail and proposes a design approach for some of them.
+
+In Gardener, scheduling shoots onto seeds is quite similar to scheduling pods onto nodes in Kubernetes. Therefore, a guiding principle behind the proposed design approaches is taking advantage of best practices and existing components already used in Kubernetes. 
+
+## Ensuring Seeds Capacity for Shoots Is Not Exceeded
+
+Seeds have a practical limit of how many shoots they can accommodate. Exceeding this limit is undesirable as the system performance will be noticeably impacted. Therefore, it is important to ensure that a seed's capacity for shoots is not exceeded by introducing a maximum number of shoots that can be scheduled onto a seed and making sure that it is taken into account by the scheduler. 
+
+An initial discussion of this topic is available in [Issue #2938](https://github.com/gardener/gardener/issues/2938). The proposed solution is based on the following flow:
+
+* The `gardenlet` is configured with certain *resources* and their total *capacity* (and, for certain resources, the amount reserved for Gardener).
+* The `gardenlet` seed controller updates the Seed status with the capacity of each resource and how much of it is actually available to be consumed by shoots, using `capacity` and `allocatable` fields that are very similar to the corresponding fields in [the Node status](https://github.com/kubernetes/api/blob/2c3c141c931c0ab1ce1396c3152c72852b3d37ee/core/v1/types.go#L4582-L4593).
+* When scheduling shoots, `gardener-scheduler` is influenced by the remaining capacity of the seed. In the simplest possible implementation, it never schedules shoots onto a seed that has already reached its capacity for a resource requested by the shoot.
+
+Initially, the only resource considered would be the maximum number of shoots that can be scheduled onto a seed. Later, more resources could be added, as well as requests for such resources by shoots. For example, if a shoot requests (via a `resourceRequirements` field) a certain number of persistent volumes, but the seed's allocatable amount of persistent volumes minus the number of persistent volumes already requested by other shoots scheduled onto that seed is lower, the shoot can't be scheduled onto that seed.
```

Comment:
Referencing some concerns that I stated in the implementation PR https://github.com/gardener/gardener/pull/3078#issuecomment-720275721 (and following)

Reply:
/cc @mvladev We really need some good motivation for having seed resources requested by shoots, or I think the entire idea is a good candidate for being dropped.

Reply:
Since there was no further feedback, removed it from the document.